### PR TITLE
chore: use userInterestsConnection to fetch collected artists

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-180] 

Keeping this as a draft because I see a PR doing the same thing [here](https://github.com/artsy/eigen/pull/9091). I thought the ticket we have in the board is only MP. cc @olerichter00 

### Description

This PR updates instances where we are using collected in favor of using `userInterestsConnection` 

https://github.com/artsy/eigen/assets/11945712/c83ee987-43e4-4959-835e-27758ebb51f8

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-180]: https://artsyproduct.atlassian.net/browse/ONYX-180?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1869b073f6ad6d9c14f6cf97ccf01cf38a358248
</p>

File: src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtistsAutosuggest.tsx [hunk 1]
```diff
@@ -167,28 +168,38 @@ const myCollectionAddCollectedArtistsAutosuggestQuery = graphql`
...
-
-  const keywordFilterWords = normalizedKeywordFilter.split(" ")
-
-  const doAllKeywordFiltersMatch = (artist: { displayLabel: string | null }) =>
-    keywordFilterWords.filter((word) => !normalizeText(artist?.displayLabel ?? "").includes(word))
-      .length === 0
-
-  return artists.filter(doAllKeywordFiltersMatch)
-}
+// export const filterArtistsByKeyword = (
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtistsAutosuggest.tsx:171]: Good idea to use the already existing method 👍

**Comment by olerichter00** [comment]
We can probably close this PR once https://github.com/artsy/eigen/pull/9091 is merged.
I also included your fix about the duplicated `filterArtistsByKeyword` function: https://github.com/artsy/eigen/pull/9091/commits/963b2838ebf57ee853a4276a9ab9e91838829ec4.

**Comment by MounirDhahri** [comment]
Great
Closing this now

**Pull request closed without merging**